src/charm.py

```diff
@@ -1,103 +1,292 @@
+            postgres_plugin=self._db_plugin,
+            ignore_capabilities=str(self.config["ignore_capabilities"]).lower(),
+            limited_failed_binds=str(self.config["limit_failed_binds"]).lower(),
+            number_of_failed_binds=self.config["number_of_failed_binds"],
+            period_of_failed_binds=self.config["period_of_failed_binds"],
+            block_failed_binds_for=self.config["block_failed_binds_for"],
+            prune_source_tables_every=self.config["prune_source_table_every"],
+            prune_sources_older_than=self.config["prune_sources_older_than"],
+            baseDN=self._basedn,
+        )
+        return rendered
+
+    def _get_database_relation_info(self) -> dict:
+        """Get database info from relation data bag."""
+        relation_id = self.database.relations[0].id
+        relation_data = self.database.fetch_relation_data()[relation_id]
+
+        return {
+            "username": relation_data.get("username"),
+            "password": relation_data.get("password"),
+            "endpoints": relation_data.get("endpoints"),
+            "database_name": self._db_name,
+        }
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to glauth container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to glauth container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+
+        if not self.model.relations[self._db_relation_name]:
+            self.unit.status = BlockedStatus("Missing required relation with postgresql")
+            event.defer()
+            return
+
+        if not self.database.is_resource_created():
+            self.unit.status = WaitingStatus("Waiting for database creation")
+            event.defer()
+            return
+
+        self._container.push(self._config_file_path, self._render_conf_file(), make_dirs=True)
+        try:
+            self._container.restart(self._container_name)
+        except ChangeError as err:
+            logger.error(str(err))
+            self.unit.status = BlockedStatus("Failed to restart, please consult the logs")
+            return
+
+        self.unit.status = ActiveStatus()
+
+    def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Event Handler for pebble ready event."""
+        # Necessary directory for log forwarding
+        if not self._container.can_connect():
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting to connect to glauth container")
+            return
+        if not self._container.isdir(str(self._log_dir)):
+            self._container.make_dir(path=str(self._log_dir), make_parents=True)
+            logger.info(f"Created directory {self._log_dir}")
```

Comment:
Could you move the comment to this block?
```suggestion
        if not self._container.isdir(str(self._log_dir)):
            # Create a directory for log forwarding
            self._container.make_dir(path=str(self._log_dir), make_parents=True)
            logger.info(f"Created directory {self._log_dir}")
```